aiida_quantumespresso/workflows/pw/relax.py

```diff
@@ -334,20 +342,16 @@ def results(self):
         else:
             self.report('maximum number of meta convergence iterations exceeded')
 
-        # Get the latest workchain, which is either the workchain_scf if it ran or otherwise the last regular workchain
-        try:
-            workchain = self.ctx.workchain_scf
-        except AttributeError:
-            workchain = self.ctx.workchains[-1]
-
-        self.out_many(self.exposed_outputs(workchain, PwBaseWorkChain))
-
-        if RelaxType(self.inputs.relax_type) is not RelaxType.NONE:
-            try:
-                structure = workchain.outputs.output_structure
-            except AttributeError:
-                structure = workchain.inputs.pw__structure
-            self.out('output_structure', structure)
+        # Get the latest relax workchain and pass the outputs
+        final_relax_workchain = self.ctx.workchains[-1]
+
+        if self.ctx.relax_type is not RelaxType.NONE:
+            self.out('output_structure', final_relax_workchain.outputs.output_structure)
+
+        if 'final_scf_inputs' in self.ctx:
+            self.out_many(self.exposed_outputs(self.ctx.workchain_scf, PwBaseWorkChain))
+        else:
+            self.out_many(self.exposed_outputs(final_relax_workchain, PwBaseWorkChain))
```

Comment:
I think this logic _should_ in principle work, but would it not be safer to just use
```
try:
    self.out_many(self.exposed_outputs(self.ctx.workchain_scf, PwBaseWorkChain))
except AttributeError:
    self.out_many(self.exposed_outputs(final_relax_workchain, PwBaseWorkChain))
```